engines/enginetest/artifacts.go

```diff
@@ -121,10 +121,11 @@ func (c *ArtifactTestCase) TestExtractNestedFolderPath() {
 
 	// Check that NestedFolderFiles was found
 	m.Lock()
+	defer m.Unlock()
 	for _, f := range c.NestedFolderFiles {
 		found := false
 		for _, f2 := range files {
-			if f == f2 {
+			if strings.HasSuffix(f2, f) {
```

Comment:
I disagreed with this... it must be the complete relative path.

Reply:
It must be the case that extracting `/folder` returns the file `/folder/hello.txt` with the name reported as `"hello.txt"`.

If you don't think this is correct, then we should change `enginetest`.

We must conform to the interface specification:
https://github.com/taskcluster/taskcluster-worker/blob/058b5b9b962078c6a1c93281ce386040f63868a1/engines/resultset.go#L14-L16

---
If you want to change the interface specification, that's a different story. Perhaps this is a bug in the artifact plugin?

Based on: https://github.com/taskcluster/taskcluster-worker/blob/master/plugins/artifacts/artifacts.go#L426
I would say it expects the `p` to be a relative path, that doesn't include slashes...
But `a.Name` doesn't end with a slash, according to: https://github.com/taskcluster/taskcluster-worker/blob/master/plugins/artifacts/payloadschema.go#L47-L55

Reply:
I am confused then, the path I am getting is absolute